test/promql/tests/data_plane/exporter_test.yaml

```diff
@@ -0,0 +1,64 @@
+evaluation_interval: 1m
+
+rule_files:
+  - prometheus.exporter_alerts.yaml
+
+tests:
+  - interval: 1m
+    input_series:
+      - series: 'konflux_up{namespace="application-service", source_cluster="cluster01"}'
+        values: '1x15'
+      - series: 'metric_up{namespace="some-service", source_cluster="cluster01"}'
+        values: '0x15'
+
+    alert_rule_test:
+      - eval_time: 15m
+        alertname: KonfluxUpAlert
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: appstudio-grafana-datasource-exporter
+            exp_annotations:
+              summary: >-
+                Exporter metric 'konflux_up' is missing.
+              description: >-
+                The Prometheus exporter metric 'konflux_up' for namespace
+                "appstudio-grafana-datasource-exporter" on cluster  
+                is missing for more than 30 minutes.
```

Comment:
The cluster label is mentioned on the rule, but you don't get it printed here as there's no time series to take labels from.
This actually raises an issue here. Those alert rules are deployed in RHOBS, where we have data coming from all clusters. It means that this alert will only fire if the metric is missing on all clusters, which is not what we want (we want it to alert as soon as it's missing from any cluster).

Try thinking if there's something we can do about it (I don't have a solution at the moment).

Reply:
(Thinking out loud)
Suppose that we somehow configure a recording rule on each cluster with the absence metric computed for the any combinations of labels for the konflux_up metric there (e.g. `absence_konflux_up`), then we should get a reading for that recording rule per-check&service, per-cluster once `konflux_up` is missing for any such combination on any cluster.

In that case, we can have a single alerting rule in RHOBS that will alert as soon as any time series in `absence_konflux_up` is giving a reading of `1`.

I think it should work in theory, but we don't have a mechanism in place for defining recording rules on clusters themselves.